homeassistant/components/myuplink/binary_sensor.py

```diff
@@ -65,11 +77,37 @@ async def async_setup_entry(
                         unique_id_suffix=point_id,
                     )
                 )
+
+    # Setup device bound sensors
+    for system in coordinator.data.systems:
+        for device in system.devices:
+            entities.append(
+                MyUplinkDeviceBinarySensor(
+                    coordinator=coordinator,
+                    device_id=device.id,
+                    entity_description=CONNECTED_BINARY_SENSOR_DESCRIPTION,
+                    unique_id_suffix="connection_state",
+                )
+            )
+
+    # Setup system bound sensors
+    for system in coordinator.data.systems:
+        device_id = system.devices[0].id
```

Comment:
Why do you need the `device_id `in a system entity? Maybe I'm overseeing something.

Reply:
This sensor is added to the HA device with this device_id. This is the most user-friendly way to handle a system with one single device. The vast majority of heatpump systems has this configuration. In a later PR we will create a separate HA device on the system level if the system contains more than one heatpump device.